src/Sylius/Bundle/ResourceBundle/Controller/ResourceController.php

```diff
@@ -238,6 +238,48 @@ public function indexAction(Request $request): Response
      *
      * @return Response
      */
+    public function massDeleteAction(Request $request): Response
+    {
+        $configuration = $this->requestConfigurationFactory->create($this->metadata, $request);
+
+        $this->isGrantedOr403($configuration, ResourceActions::MASS_DELETE);
+        /** @var array $resources */
+        $resources = $this->resourcesCollectionProvider->get($configuration, $this->repository);
+
+        foreach($resources as $resource) {
+            $event = $this->eventDispatcher->dispatchPreEvent(ResourceActions::DELETE, $configuration, $resource);
+
+            if ($event->isStopped() && !$configuration->isHtmlRequest()) {
+                throw new HttpException($event->getErrorCode(), $event->getMessage());
```

Comment:
Should it actually do an exception here? In this case only part of the data will be removed, if there is some condition on the resource that might prevent deletion on some of them, but you need to continue further anyway? Or maybe it needs to be transnational - everything or nothing.